core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3283,6 +3283,9 @@ class KafkaApis(val requestChannel: RequestChannel,
   def handleAlterIsrRequest(request: RequestChannel.Request): Unit = {
     val zkSupport = metadataSupport.requireZkOrThrow(KafkaApis.shouldNeverReceive(request))
     val alterIsrRequest = request.body[AlterIsrRequest]
+    if (config.liDenyAlterIsr) {
+      throw new ClusterAuthorizationException(s"Request $request is not authorized.")
```

Comment:
Can we make the error message more clear, i.e., "The AlterIsrRequest is denied in controller due to config liDenyAlterIsr is set to true, which is not expected in Prod environment. If this happens, update li.deny.alter.isr to false"?

Reply:
Good suggestion. Updated.